lib/instrumentation/when/nr-hooks.js

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const instrumentation = require('./index')
+
+module.exports = [
+  {
+    type: null,
+    moduleName: 'when',
+    onRequire: instrumentation
```

Comment:
I thought we only needed `nr-hooks` when an `onResolved` handler was needed?

Reply:
we also need nr-hooks when you want to register instrumentation that does not match the resolved name.  It will look for `lib/instrumentation/when.js`.  So you have to update instrumentations to pass in "module" in this case